src/randoop/reflection/ReflectionManager.java

```diff
@@ -107,8 +107,28 @@ public void apply(Class<?> c) {
     
   }
 
-
-
+  /**
+   * Applies the visitors to the methods of the class. 
+   * Will visit all methods unless visiting only public members.
+   * 
+   * @param c  the class whose methods should be visited
+   */
+  private void applyMethods(Class<?> c) {
+    Set<Method> methods = new HashSet<>();
```

Comment:
Thanks for spotting that.  I will check into what is going on with the tests.  My test should have caught that.

I don't understand what you are thinking about in terms of a single loop.  The method Class.getMethods() returns all of the public methods of the class, and the method Class.getDeclaredMethods() returns the methods declared within the class (not just public). There is no way to "get all methods", so to obtain public + not-private both methods have to be used.